libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/PathLookupImpl.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.entitlement.runtime.policy;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import static java.util.Objects.requireNonNull;
+
+public record PathLookupImpl(
+    Path homeDir,
+    Path configDir,
+    Path[] dataDirs,
+    Path[] sharedRepoDirs,
+    Path libDir,
+    Path modulesDir,
+    Path pluginsDir,
+    Path logsDir,
+    Path tempDir,
+    Path pidFile,
+    Function<String, Stream<String>> settingResolver
+) implements PathLookup {
+
+    public PathLookupImpl {
+        requireNonNull(homeDir);
+        requireNonNull(dataDirs);
+        if (dataDirs.length == 0) {
+            throw new IllegalArgumentException("must provide at least one data directory");
+        }
+        requireNonNull(sharedRepoDirs);
+        requireNonNull(configDir);
+        requireNonNull(libDir);
+        requireNonNull(modulesDir);
+        requireNonNull(pluginsDir);
+        requireNonNull(logsDir);
+        requireNonNull(tempDir);
+        requireNonNull(settingResolver);
+    }
+
+    @Override
+    public Stream<Path> getBaseDirPaths(BaseDir baseDir) {
+        if (baseDir == BaseDir.HOME) {
```

Comment:
Could this be a switch on baseDir? Then there is no need for the else case because we can ensure all values of the enum are covered.

Reply:
Changed.